src/aten/sycl/ReflectionPadKernels.cpp

```diff
@@ -0,0 +1,411 @@
+    int64_t output_xy,
+    const sycl::nd_item<3> item) {
+  // 3D grid of 1D blocks
+  auto input_offset =
+      (item.get_group(1) + item.get_group(2) * item.get_group_range(1)) *
+      input_dim_x * input_dim_y;
+  auto output_offset =
+      (item.get_group(1) + item.get_group(2) * item.get_group_range(1)) *
+      output_dim_x * output_dim_y;
+
+  auto output_x = output_xy % output_dim_x;
+  auto output_y = output_xy / output_dim_x;
+
+  auto i_start_x = std::max(int64_t(0), -pad_l);
+  auto i_start_y = std::max(int64_t(0), -pad_t);
+  auto o_start_x = std::max(int64_t(0), pad_l);
+  auto o_start_y = std::max(int64_t(0), pad_t);
+
+  int64_t input_x = std::abs(output_x - pad_l) -
+      std::abs(output_x - (input_dim_x + pad_l - 1)) - output_x + 2 * pad_l +
+      input_dim_x - 1 - o_start_x + i_start_x;
+
+  int64_t input_y = std::abs(output_y - pad_t) -
+      std::abs(output_y - (input_dim_y + pad_t - 1)) - output_y + 2 * pad_t +
+      input_dim_y - 1 - o_start_y + i_start_y;
+
+  return std::make_pair<int64_t, int64_t>(
+      input_offset + input_y * input_dim_x + input_x,
+      output_offset + output_y * output_dim_x + output_x);
+}
+
+template <typename scalar_t>
+struct ReflectionPad2dOutKernellFunctor {
+  void operator()(sycl::nd_item<3> item) const {
+    auto output_xy = item.get_global_id(0);
+
+    if (output_xy < output_dim_x_ * output_dim_y_) {
+      // input index and output index mapping
+      auto index_pair = get_index_mapping2d(
+          input_dim_x_,
+          input_dim_y_,
+          output_dim_x_,
+          output_dim_y_,
+          pad_l_,
+          pad_t_,
+          output_xy,
+          item);
+      output_[index_pair.second] = input_[index_pair.first];
+    }
+  }
+  ReflectionPad2dOutKernellFunctor(
+      scalar_t* input,
+      scalar_t* output,
+      int64_t input_dim_x,
+      int64_t input_dim_y,
+      int64_t pad_t,
+      int64_t pad_l,
+      int64_t output_dim_x,
+      int64_t output_dim_y)
+      : input_(input),
+        output_(output),
+        input_dim_x_(input_dim_x),
+        input_dim_y_(input_dim_y),
+        pad_t_(pad_t),
+        pad_l_(pad_l),
+        output_dim_x_(output_dim_x),
+        output_dim_y_(output_dim_y) {}
+
+ private:
+  scalar_t* input_;
+  scalar_t* output_;
+  int64_t input_dim_x_;
+  int64_t input_dim_y_;
+  int64_t pad_t_;
+  int64_t pad_l_;
+  int64_t output_dim_x_;
+  int64_t output_dim_y_;
+};
+
+template <typename scalar_t>
+void reflection_pad2d_out_kernel_impl(
+    scalar_t* input,
+    scalar_t* output,
+    int64_t input_dim_x,
+    int64_t input_dim_y,
+    int64_t pad_t,
+    int64_t pad_b,
+    int64_t pad_l,
+    int64_t pad_r,
+    int64_t nbatch,
+    int64_t nplane) {
+  int64_t output_dim_x = input_dim_x + pad_l + pad_r;
+  int64_t output_dim_y = input_dim_y + pad_t + pad_b;
+
+  auto queue = at::xpu::getCurrentSYCLQueue();
+  int work_group_size =
+      output_dim_x * output_dim_y > 256 ? 256 : output_dim_x * output_dim_y;
```

Comment:
256 is not compatible. I might think it followed stock CUDA hard-coded. Can we evaluate it what we should use under SYCL?